src/REPLMode.jl

```diff
@@ -167,7 +167,7 @@ end
 ################
 # REPL parsing #
 ################
-const lex_re = r"^[\?\./\+\-](?!\-) | [^@\#\s;]+\s*=\s*[^@\#\s;]+ | \#\s*[^@\#\s;]* | @\s*[^@\#\s;]* | [^@\#\s;]+|;"x
+const lex_re = r"^[\?\./\+\-](?!\-) | ((git|ssh|http(s)?)|(git@[\w\.]+))(:(//)?)([\w\.@\:/\-~]+)(\.git)(/)? | [^@\#\s;]+\s*=\s*[^@\#\s;]+ | \#\s*[^@\#\s;]* | @\s*[^@\#\s;]* | [^@\#\s;]+|;"x
```

Comment:
This regex is becoming quite a beast :D. Oh well.

Reply:
It has an `x` modifier so it can be split across lines.

Reply:
Feel free to take inspiration from: https://github.com/JuliaLang/julia/blob/master/stdlib/LibGit2/src/utils.jl#L5-L26